hathor/indexes/memory_tx_group_index.py

```diff
@@ -57,13 +57,14 @@ def remove_tx(self, tx: BaseTransaction) -> None:
         assert tx.hash is not None
 
         for key in self._extract_keys(tx):
-            self.index[key].discard(tx.hash)
+            self.index[key].discard((tx.timestamp, tx.hash))
 
     def _get_from_key(self, key: KT) -> Iterable[bytes]:
-        yield from self.index[key]
+        for _, h in self.index[key]:
+            yield h
 
     def _get_sorted_from_key(self, key: KT) -> Iterable[bytes]:
-        return sorted(self.index[key])
+        return [h for t, h in sorted(self.index[key])]
```

Comment:
```suggestion
        return [h for _, h in sorted(self.index[key])]
```

Reply:
This reminds me that I wanted to find some linter that is able to alert about unused variables. Sometimes they're the source of bugs and we have a lot of code that we could clean up with this. Our current setup doesn't catch it.